test/e2e/testsuites/dynamically_provisioned_pod_affinity.go

```diff
@@ -67,6 +67,15 @@ func (t *PodNodeAffinity) Run(client clientset.Interface, namespace *v1.Namespac
 		labelCleanup, err := SetNodeLabels(client, nodeObj, testLabel)
 		framework.ExpectNoError(err)
 		defer labelCleanup()
+
+		// if the node is a master node, it will not have the required node affinity to schedule pod.
+		// so add the label
+		if nodeObj.Labels != nil && nodeObj.Labels[nodeRoleLabel] == "master" {
+			roleLabelCleanup, err := SetNodeLabels(client, nodeObj, map[string]string{driver.TopologyKey: ""})
```

Comment:
So it the label `topology.disk.csi.azure.com/zone` is not set if there is only one master node regardless of whether its multi-az or not. Setting to to "" is the correct value for the single-az case, but in multi-az it should be set to something like "westus2-0". The correct value could be derived from the `topology.kubernetes.io/region` and `topology.kubernetes.io/zone` values which seem to be set regardless.